modules/smartadserverBidAdapter.js

```diff
@@ -82,6 +82,27 @@ export const spec = {
         payload.gdpr_consent = bidderRequest.gdprConsent.consentString;
         payload.gdpr = bidderRequest.gdprConsent.gdprApplies; // we're handling the undefined case server side
       }
+	  
+	  if (bid && bid.userId) {
+        for (let source in bid.userId) {
+          let foundId = '';
+          let obj = bid.userId[source];
+          if (source === 'digitrustid' && obj && typeof obj === 'object' && obj.data && obj.data.id) {
+            foundId = obj.data.id;
+          } else if (obj && typeof obj === 'string') {
+            foundId = obj;
+          }
+          if (foundId !== '') {
+            if (payload.eids === undefined) {
+              payload.eids = [];
+            }
+            payload.eids.push({
```

Comment:
why not just use `userIdsAsEids` instead, that way the identity providers can handle the formatting for you without all the extra code above?

Reply:
Excellent point, now i use createEidsArray found in eid.js, works a treat.